### PR TITLE
Constrain token payment by auth entries

### PR DESCRIPTION
**Pull request by aristidesstaffieri**
Adds checks during token simulation to attempt to constrain simulation to a typical transfer implementation.

### CODE REVIEW

File: src/route/index.ts [hunk 1]
```diff
@@ -549,17 +551,48 @@ export async function initApiServer(
...
+            for (const auth of auths) {
+              if (
+                auth.credentials().switch() !==
+                xdr.SorobanCredentialsType.sorobanCredentialsSourceAccount()
+              ) {
+                throw new Error(ERROR.ACCOUNT_NOT_SOURCE);
+              }
+              if (
+                auth.credentials().switch() ===
+                  xdr.SorobanCredentialsType.sorobanCredentialsAddress() &&
```
Comment (sisuresh) [inline src/route/index.ts:574]: Is it possible for this condition to be true? The previous check will throw if the credential is not `sorobanCredentialsSourceAccount` right?

Comment (aristidesstaffieri) [reply src/route/index.ts:574]: ah yeah thanks for pointing that out. This shouldn't be needed at all then, I'll remove it.

Comment (aristidesstaffieri) [reply src/route/index.ts]: removed in 2b943fbac28c06eb9add751f3477c109aadcd89f

File: src/helper/error.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const ERROR = {
+  ACCOUNT_NOT_SOURCE:
+    "Transfer contains authorization entry for a different account",
+  FOREIGN_CONTRACT_AUTH:
```
Comment (mtwtfss) [inline src/helper/error.ts:4]: did we mean to use this error in this pull request or are you planning to use it in a follow-up change?

File: src/route/index.ts [hunk 3]
```diff
@@ -545,17 +547,37 @@ export async function initApiServer(
...
             )) as SorobanRpc.Api.SimulateTransactionSuccessResponse;
+
             const preparedTransaction = SorobanRpc.assembleTransaction(
               tx,
               simulationResponse
             );
+
+            const built = preparedTransaction.build();
+            const sorobanOp = built
+              .operations[0] as Operation.InvokeHostFunction;
```
Comment (mtwtfss) [inline src/route/index.ts:558]: should we inspect all operations or is only looking at the first one always sufficient?

File: src/helper/error.ts (hunk 2 above)
Comment (aristidesstaffieri) [reply src/helper/error.ts:4]: Actually I'll remove that one, @sisuresh pointed how that check was already covered.

File: src/route/index.ts (hunk 3 above)
Comment (aristidesstaffieri) [reply src/route/index.ts:558]: In the case of a tx with an InvokeHostFunction, there can only be 1 operation. We could check for the operation type here but only the InvokeHostFunction type has auth. I'll add an explicit check for the op type though, that seems a bit more clear.

File: src/helper/error.ts (hunk 2 above)
Comment (aristidesstaffieri) [reply src/helper/error.ts]: removed in 3272c3c01df00ef3db048298a44055acca7e0b97

File: src/route/index.ts (hunk 3 above)
Comment (aristidesstaffieri) [reply src/route/index.ts]: Added an explicit check for op type in 3272c3c01df00ef3db048298a44055acca7e0b97

**Pull request merged**